PeerioTechnologies/peerio-desktop
# feat: url preview

#### Relevant info and issue/PR links
https://app.clubhouse.io/peerio/story/17377/desktop-url-preview
icebear: https://github.com/PeerioTechnologies/peerio-icebear/pull/359

#### Testing instructions
URL previews working! The Figma mockups are linked in Clubhouse but there's a lot more in there than what we've implemented for this first MVP. Here are the possible scenarios:

1. direct link to an image file
  * show :framed_picture: icon 
  * show "Expand/collapse preview" text, which is clickable, as well as toggle button
  * preview just shows the image file, no website name etc.

2. link to website
  * show website's favicon (if available), otherwise :globe_with_meridians:  icon
  * show website's name (if available), otherwise hostname
  * if there is preview content, show expand/collapse toggle
  * preview content shows page title, description, and image

3. check that there are no CSP errors in console as in https://app.clubhouse.io/peerio/story/14391/url-unfurl-tries-to-load-css-blocked-by-content-security-policy (this PR has a different method of fixing that issue, the previous method was removed.)

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
